File: WEEK 1 c#/Square.cs
//Square 

using System;

public class Square
{
    private static Square instance;
    private int side;

    private Square(int side)
    {
        this.side = side;
    }

    public static Square GetInstance(int side)
    {
        if (instance == null)
        {
            instance = new Square(side);
        }
        return instance;
    }

    public int GetArea()
    {
        return side * side;
    }

    public int GetSide()
    {
        return side;
    }

    public void SetSide(int side)
    {
        this.side = side;
    }

    public static void Main(string[] args)
    {
        Square square1 = Square.GetInstance(5);
        Console.WriteLine("Square1 side: " + square1.GetSide());
        Console.WriteLine("Square1 area: " + square1.GetArea());

        Square square2 = Square.GetInstance(10); 
        Console.WriteLine("Square2 side: " + square2.GetSide()); 
        Console.WriteLine("Square2 area: " + square2.GetArea());

        Console.WriteLine("square1 == square2? " + (square1 == square2));
    }
}

File: WEEK 1/Square.java
//Singleton design

public class Square{
    private static Square instance;
    private int side;
    private Square(int side) {
        this.side = side;
    }
    public static Square getInstance(int side) {
        if (instance == null) {
            instance = new Square(side);
        }
        return instance;
    }
    public int getArea() {
        return side * side;
    }
    public int getSide() {
        return side;
    }
    public void setSide(int side) {
        this.side = side;
    }
    public static void main(String[] args) {
        Square square1 = Square.getInstance(5);
        System.out.println("Square1 side: " + square1.getSide());    
        System.out.println("Square1 area: " + square1.getArea());        
        Square square2 = Square.getInstance(10);
        System.out.println("Square2 side: " + square2.getSide());        
        System.out.println("Square2 area: " + square2.getArea());        
        System.out.println("square1 == square2? " + (square1 == square2)); 
    }
}